### DISCUSSION TITLE
Edit a python expression

### DISCUSSION THREAD

**Discussion by jazzabeanie** [saulpw/visidata#1314] [Q&A] [answered]
Let's say I create a new column with `=`. How do I then edit the expression that populates the column? I know I could recreate it, but if I've created other columns that depend on the column that I want to change, then it becomes painful.

**Comment by geekscrapy** [comment, marked as answer]
You can open ColumnsSheet (SHIFT+C) then you'll see a column which is expr. Use `e` to edit the value in this column to the correct expression!

**Comment by saulpw** [comment]
@geekscapy is correct.  Also, column bindings are based on the column name, so if you make a new column, give it the same name, and *delete* the old column (or at least move it to the end of the columns list; just hiding it isn't enough), then your other non-cached columns should update automatically.  (and `gz'` will reset the cache for all visible columns, though you won't need to do that in most cases).

**Reply by jazzabeanie** [reply]
Ok, that's nice. And if I have two column names with the same name, how do I reference the later one by var name?

**Reply by geekscrapy** [reply]
If you use a column name within an expression and there are multiple columns of the same name, it will use the leftmost column first. Column names (used in the expressions) are matched based on the order of the columns (which is a list internally), therefore, the column with the lower list index will be chosen first

**Comment by arisolt** [comment]
I'm trying to follow those instructions, but when I try to edit the `expr` column, it shows me `IndexError: list index out of range`. Any suggestions?

Also, fwiw the `expr` column in Columns Sheet is empty (even tho the column has been created successfully using `=` and a Python expression).

**Reply by saulpw** [reply]
File an issue with all the details, @arisolt.  This is nowhere near enough information to help you diagnose the issue.
